src/App.tsx

```diff
@@ -15,35 +15,101 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  name = 'name',
+  length = 'length',
+}
+
 export const App: React.FC = () => {
+  const [visibleGoods, setVisibleGoods] = useState<string[]>(goodsFromServer);
+  const [sortField, setSortField] = useState<SortType | ''>('');
+  const [isReversed, setIsReversed] = useState(false);
+
+  if (sortField) {
+    const sortedGoods = [...visibleGoods].sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.name:
+          return good1.localeCompare(good2);
+        case SortType.length:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+
+    if (isReversed) {
+      sortedGoods.reverse();
+    }
+
```

Comment:
move this logic to some function (getVisibleGoods)

and just call it
`const visibleGoods = getVisibleGoods(...........)`